changelog/unreleased/pr-20558.toml

```
@@ -0,0 +1,5 @@
+type = "c" # One of: a(dded), c(hanged), d(eprecated), r(emoved), f(ixed), s(ecurity)
+message = "UserResource.list() now works for every authenticated user. In case the calling user lacks the 'RestPermissions.USERS_LIST' permission, only the calling user is returned."
```

I think we should write this in a more user-friendly way (changelogs are not meant for developers, but for users/people consuming our API). A suggestion:
```suggestion
message = "API endpoint for listing users will not return an error anymore when user is lacking permission, but return current user only instead."
```